src/doc/sprite.cpp

```diff
@@ -368,6 +370,25 @@ RgbMap* Sprite::rgbMap(frame_t frame, RgbMapFor forLayer) const
   return m_rgbMap;
 }
 
+OctreeMap* Sprite::octreeInit(frame_t frame) const
+{
+  return octreeInit(frame, backgroundLayer() ? RgbMapFor::OpaqueLayer:
+                                           RgbMapFor::TransparentLayer);
+}
+
+OctreeMap* Sprite::octreeInit(frame_t frame, RgbMapFor forLayer) const
+{
+  int maskIndex = (forLayer == RgbMapFor::OpaqueLayer ?
+                   -1: transparentColor());
+
+  if (m_octreeMap == NULL || m_octreeMap->getModifications() != palette(frame)->getModifications())
+    m_octreeMap = new OctreeMap();
```

Comment:
As `m_octreeMap` is not an `std::unique_ptr`, there is a memory leak here when `m_octreeMap` is not `nullptr` and the modifications don't match. It needs to do something like:
```cpp
delete m_octreeMap;
m_octreeMap = new OctreeMap;
```
(or we can use a `std::unique_ptr` something that wasn't used for the rgbMap just because I wasn't using unique pointers at the time, I think this code is quite old)